### PR TITLE
polish(diff): add diff method accepting options for diffing AMTs with nondefault options

### PR DESCRIPTION
**Pull request by frrist**
- Allow AMT's with non-default bitwidths to be diffed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,10 +3,11 @@ module github.com/filecoin-project/go-amt-ipld/v3
 go 1.12
 
 require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/ipfs/go-block-format v0.0.2
 	github.com/ipfs/go-cid v0.0.7
 	github.com/ipfs/go-ipld-cbor v0.0.4
-	github.com/ipfs/go-log v1.0.4
+	github.com/kr/pretty v0.1.0 // indirect
```
Comment (Stebalien) [inline go.mod:10]: why update these?

**Review by Stebalien — CHANGES_REQUESTED** [review]
I'm pretty sure this doesn't work. We a least need a test.

File: go.mod (hunk 1 above)
Comment (frrist) [reply go.mod:10]: I didn't intend to, I just ran `go mod tidy` before I pushed. I think they were dirty from a past commit.

**Comment by frrist** [comment]
@Stebalien you're correct, it doesn't work (added test to make sure). Closing.

**Comment by frrist** [comment]
reopening as there is value in controlling the options an AMT node is loaded with, e.g. being able to diff 2 AMT nodes that have Bitwidths differing from the default value.

**Comment by Stebalien** [comment]
> reopening as there is value in controlling the options an AMT node is loaded with, e.g. being able to diff 2 AMT nodes that have Bitwidths differing from the default value.

This _is_ valuable, but the current method isn't the right way to go about this as we can still end up diffing AMTs with different bit-widths leading to nonsensical results. We _could_ just check inside `Diff`, but it's likely easier to just make `Diff` accept `opts ...Option` like `LoadAMT` does, forwarding the option to `LoadAMT`.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**